ui/lib/kv/addon/components/page/secret/metadata/details.js

```diff
@@ -28,19 +28,20 @@ export default class KvSecretMetadataDetails extends Component {
   @tracked deleteModalOpen = false;
   @service flashMessages;
   @service router;
+  @service store;
 
-  @action async handleDelete() {
-    // the only delete option from this view is delete on metadata.
+  @action
+  async onDelete(model) {
```

Comment:
we don't need to pass `model` in here - this could just be `this.args.metadata.destroyRecord()`